src/metalnx-web/src/main/java/com/emc/metalnx/controller/LoginController.java

```diff
@@ -143,12 +146,16 @@ public ModelAndView loginErrorHandler(final Exception e) {
 		 * see if I'm really logged in as a hacky way of getting around this bug
 		 * https://github.com/irods-contrib/metalnx-web/issues/162
 		 */
-		DataGridUser loggedInUser = loggedUserUtils.getLoggedDataGridUser();
-		if (loggedInUser != null) {
-			logger.warn("is already logged in go ahead and just hit the main page...");
-			String redirect = "redirect:/browse/home";
-			ModelAndView model = new ModelAndView(redirect);
-			return model;
+		Authentication auth = SecurityContextHolder.getContext().getAuthentication();
+		if (auth.isAuthenticated() && !(auth instanceof AnonymousAuthenticationToken)) {
+		    DataGridUser loggedInUser = loggedUserUtils.getLoggedDataGridUser();
+		
+		    if (loggedInUser != null) {
+			    logger.warn("is already logged in go ahead and just hit the main page...");
+			    String redirect = "redirect:/browse/home";
+			    ModelAndView model = new ModelAndView(redirect);
+			    return model;
+		    }
```

Comment:
What does this change achieve?

I'm specifically referring to L150.

Reply:
With the changes to remove the admin login, the getLoggedDataGridUser() actually throws an exception when there hasn't been a login.  (That is my recollection.)  So to do what is now needed I have to make sure there is a non-anonymous authentication first.

Reply:
Ok.

> ... So to do what is now needed ...

What is _needed_?

> ... I have to make sure there is a non-anonymous authentication first.

I'm interpreting this to say you're checking for the anonymous user to determine how to proceed. Is that correct?

Reply:
The updated comment in HttpResponseHandlerInterceptor.java looks like it applies to this code change too. Is that correct?

Reply:
> > ... I have to make sure there is a non-anonymous authentication first.

If nobody has logged in there is an auth instance of `AnonymousAuthenticationToken`.  In this case `loggedUserUtils.getLoggedDataGridUser()` throws an exception.  Since we don't have a real login, we do not want to redirect to `/browse/home`.

Reply:
Got it. Consider adding a comment to the code explaining that. It's likely to be useful to next reader of the code.